pkg/awsutils/awsutils_test.go

```diff
@@ -274,6 +275,60 @@ func TestGetAttachedENIs(t *testing.T) {
 	assert.Equal(t, len(ens), 2)
 }
 
+func TestGetAttachedENIsOnErr(t *testing.T) {
+	ctrl, mockMetadata, _ := setup(t)
+	defer ctrl.Finish()
+
+	mockMetadata.EXPECT().GetMetadata(metadataMACPath).Return(primaryMAC+" "+eni2MAC, nil)
+
+	gomock.InOrder(
+		mockMetadata.EXPECT().GetMetadata(metadataMACPath+primaryMAC+metadataDeviceNum).Return(eni1Device, nil),
+		mockMetadata.EXPECT().GetMetadata(metadataMACPath+primaryMAC+metadataInterface).Return(primaryMAC, nil),
+		mockMetadata.EXPECT().GetMetadata(metadataMACPath+primaryMAC+metadataSubnetCIDR).Return(subnetCIDR, nil),
+		mockMetadata.EXPECT().GetMetadata(metadataMACPath+primaryMAC+metadataIPv4s).Return("", nil),
+		mockMetadata.EXPECT().GetMetadata(metadataMACPath+eni2MAC+metadataDeviceNum).Return("", errors.New("Arbitrary error")),
+	)
+	ins := &EC2InstanceMetadataCache{ec2Metadata: mockMetadata}
+	_, err := ins.GetAttachedENIs()
+
+	assert.Error(t, err)
+}
+
+func TestGetAttachedENIsOn404(t *testing.T) {
+	ctrl, mockMetadata, _ := setup(t)
+	defer ctrl.Finish()
+
+	mockMetadata.EXPECT().GetMetadata(metadataMACPath).Return(primaryMAC+" "+eni2MAC, nil)
+
+	eniNotFoundErr := awserr.New("EC2MetadataError", "failed to make EC2Metadata request", errors.New(
```

Comment:
This error is constructed exactly [like it is in the AWS SDK](https://github.com/aws/aws-sdk-go-v2/blob/d52522b5f4b95591ff6528d7c54923951aadf099/aws/ec2metadata/service.go#L103), with the HTML copied from logs (visible in the PR message, if expanded).